src/actions/omnisharp-server-actions.el

```diff
@@ -2,37 +2,65 @@
   "Path to OmniSharp. If its value is nil, search for the server in the exec-path"
   :type '(choice (const :tag "Not Set" nil) string))
 
-;;;###autoload
-(defun omnisharp-start-omnisharp-server (path-to-project)
-  "Starts an OmniSharp server server for a given path to a project file or a directory"
-  (interactive "GStart OmniSharp for project folder or solution file: ")
-  (unless (bound-and-true-p omnisharp-server-executable-path)
-    (error "Could not find the OmniSharp executable. Please set the variable omnisharp-server-executable-path to a valid path"))
-
-  (if (or (file-directory-p path-to-project)
-          (file-exists-p path-to-project))
-      (progn
-        (message (format "Starting OmniSharpServer using project folder/solution file: %s" path-to-project))
-        (message "using the server at: %s" omnisharp-server-executable-path))
-    (error (format "Path does not lead to a valid project folder or solution file path: %s" path-to-project)))
+(defun omnisharp--do-server-start (path-to-project)
+  (message (format "Starting OmniSharpServer using project folder/solution file: %s" path-to-project))
+  (message "using the server at: %s" omnisharp-server-executable-path)
 
   ;; Save all csharp buffers to ensure the server is in sync"
   (save-some-buffers t (lambda () (string-equal (file-name-extension (buffer-file-name)) "cs")))
 
+  (setq omnisharp--last-project-path path-to-project)
```

Comment:
This is okay. I think they might be in the `omnisharp-server-info` thing we have for the running connection as well, instead of here.

But if you want, it can be changed when/if we support multiple backends at the same time.